src/components/App.tsx

```diff
@@ -1586,6 +1568,7 @@ export class App extends React.Component<any, AppState> {
         p1[0] += deltaX;
         p1[1] += deltaY;
       }
+      mutateElement(element);
```

Comment:
Were you too lazy to refactor the code to move the mutation in the second arg? Or is it the way the code is intended to be written?

Reply:
yes, i fixed this is in a follow up PR that i'll put up after this is merged.